Lore: History and cultural knowledge in the Elm community

supermario opened the pull request.

jfmengels commented:
Additional remarks on "Elm Core Development"
> New discussion points aren't brought up
I find the wording odd? Maybe that's just me or my English. Not the best wording but something like "Discussions tend to rarely bring up new discussion points" maybe? Feel free to ignore if you think it's okay.

---

One thing I often hear is "Evan should clearly communicate<something>" (his work process, core team status, his secret project, etc.). I feel like that could merit a mention, for instance in "Unhelpful ways to progress", maybe as a sub-point to the first bullet point.

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+
+---
+
+*Authored by non-core Elm community members: Mario Rogic and Jakub Hampl. Both Mario and Jakub have been in the Elm Community since 2016. Mario maintains Lamdera and Elmcraft. Jakub works on elm-visualization. This page expresses our private opinions; however we have some indication that these ideas are not uncommon among the more seasoned members of the community. Editors: Jeroen Engels, Wolfgan ...
```
```suggestion
*Authored by non-core Elm community members: Mario Rogic and Jakub Hampl. Both Mario and Jakub have been in the Elm Community since 2016. Mario maintains [Lamdera](https://lamdera.com) and Elmcraft. Jakub works on [elm-visualization](https://package.elm-lang.org/packages/gampleman/elm-visualization/latest/). This page expresses our private opinions; however we have some indication that these ideas are not uncommon among the more seasoned members of the community. Editors: Jeroen Engels, Wolfgang Schuster.*
```

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+- [The Elm Compiler](https://github.com/elm/compiler)
+- [https://elm-lang.org/]() - [github](https://github.com/elm/elm-lang.org)
+- [https://package.elm-lang.org/](https://package.elm-lang.org/) - github
+- The `elm/*` Elm packages: [elm/core](), [elm/browser](), [elm/bytes](), [elm/file](), [elm/html](), [elm/http](), [elm/json](), [elm/parser](), [elm/random](), [elm/regex](), [elm/svg](), [elm/time](), [elm/url](), elm/virtual-dom, elm-explorations/markdown
```
```suggestion
- The `elm/*` Elm packages: [elm/core](https://package.elm-lang.org/packages/elm/core/latest/), [elm/browser](https://package.elm-lang.org/packages/elm/browser/latest/), [elm/bytes](https://package.elm-lang.org/packages/elm/bytes/latest/), [elm/file](https://package.elm-lang.org/packages/elm/file/latest/), [elm/html](https://package.elm-lang.org/packages/elm/html/latest/), [elm/http](https://package.elm-lang.org/packages/elm/http/latest/), [elm/json](https://package.elm-lang.org/packages/elm/json/latest/), [elm/parser](https://package.elm-lang.org/packages/elm/parser/latest/), [elm/random](https://package.elm-lang.org/packages/elm/random/latest/), [elm/regex](https://package.elm-lang.org/packages/elm/regex/latest/), [elm/svg](https://package.elm-lang.org/packages/elm/svg/latest/), [elm/time](https://package.elm-lang.org/packages/elm/time/latest/), [elm/url](https://package.elm-lang.org/packages/elm/url/latest/), [elm/virtual-dom](https://package.elm-lang.org/packages/elm/virtual-dom/latest/), [elm-explorations/markdown](https://package.elm-lang.org/packages/elm-explorations/markdown/latest/)
```

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+- [https://package.elm-lang.org/](https://package.elm-lang.org/) - github
+- The `elm/*` Elm packages: [elm/core](), [elm/browser](), [elm/bytes](), [elm/file](), [elm/html](), [elm/http](), [elm/json](), [elm/parser](), [elm/random](), [elm/regex](), [elm/svg](), [elm/time](), [elm/url](), elm/virtual-dom, elm-explorations/markdown
+
+All these projects (the “Elm core projects”) are released with an open source licence (BSD-3-Clause).
```
```suggestion
All these projects (the “Elm core projects”) are released with an open source licence ([BSD-3-Clause](https://github.com/elm/core/blob/master/LICENSE)).
```

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+    - [Lamdera](https://lamdera.com), a backwards/forwards-compatible soft-fork of the Elm compiler, with features for full-stack Elm and extensions for Elm tooling authors and Elm users
+    - [elm-pages](https://elm-pages.com) - JAMstack for elm.
+  - Some people take inspiration and start new initiatives not compatible with Elm core
+    - Gren, a hard fork of Elm by Robin Heggelund Hansen, previous contributor to various Elm projects
```
```suggestion
    - [Gren](https://gren-lang.org), a hard fork of Elm by Robin Heggelund Hansen, previous contributor to various Elm projects
```

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+
+- That Evan has never had these conversations before
+- That because Evan is obviously wrong, the only remaining issue is to badger him until he becomes rational and changes his mind
+
```
This feels like its missing a conclusion.

```suggestion

Since the second has been tried a lot (unsuccessfully), we can safely conclude the first isn't the case.
```

gampleman commented on content/lore/elm-core-development.md:
```diff
@@ -0,0 +1,186 @@
+
+### But surely this way of doing things will cause people to abandon Elm? How is Elm going to achieve wide adoption?
+
+There is a certain inconsistency between some early Elm material (notably “Let’s be mainstream!”) and some of the later material (notably “The hard parts of Open Source”). Elm in the more distant past indeed was - and promoted itself to be - more open to participation, and a lot of this was scaled back over time. Humans are like that - we change. Our situations change. Our opinions and the tradeo ...
```
```suggestion
There is a certain inconsistency between some early Elm material (notably [“Let’s be mainstream!”](https://www.youtube.com/watch?v=oYk8CKH7OhE)) and some of the later material (notably [“The hard parts of Open Source”](https://www.youtube.com/watch?v=o_4EX4dPppA)). Elm in the more distant past indeed was - and promoted itself to be - more open to participation, and a lot of this was scaled back over time. Humans are like that - we change. Our situations change. Our opinions and the tradeoffs we are willing to make evolve as we get older and get to know ourselves.
```

gampleman reviewed (approved):
Looks good, just added some missing hyperlinks.

jfmengels commented:
> One thing I often hear is "Evan should clearly communicate" (his work process, core team status, his secret project, etc.). I feel like that could merit a mention, for instance in "Unhelpful ways to progress", maybe as a sub-point to the first bullet point.

Some of my comments have been addressed, but I don't think this one has been. Just wanted to check whether that was on purpose (because you think it's better not to or because it's already addressed enough in your opinion) or not.

supermario commented:
Thanks @jfmengels – improved that wording, added the note re markdown package.

Re communication, I added an [explicit page to address what Evan is working on](https://deploy-preview-19--elmcraft.netlify.app/lore/what-is-evan-working-on/). I figured that was better than to start getting into addressing specific "Evan should x"'s, because I think even that specific point still falls under the "Yes, people agree, but that's not how he's doing things".

Janiczek commented:
@supermario I know I'm like a broken record on this question, but how can these articles be reached from elmcraft.org? 😅

supermario commented:
@Janiczek yeah sorry fixed in https://github.com/elmcraft/elmcraft.org/commit/fae150e3bd1477b7aa578c9085df09c95d1a45ea#diff-582c007730af5767a48f16cb5632234d7dfac36a12e16ae9ff6e40bba2a11a34R74 :)
